tests/samples/cmake-files/targetLinkLibrariesWithDynamicLookup.cmake

```diff
@@ -0,0 +1,278 @@
+        result = 1;
+
+      done:
+        if(counter_module) dlclose(counter_module);
+        return result;
+    }
+  ")
+
+  set(_rpath_arg)
+  if(APPLE AND ${CMAKE_VERSION} VERSION_GREATER 2.8.11)
+    set(_rpath_arg "-DCMAKE_MACOSX_RPATH='${CMAKE_MACOSX_RPATH}'")
+  endif()
+
+  try_compile(project_compiles
+              "${test_project_bin_dir}"
+              "${test_project_src_dir}"
+              "${projectName}"
+              CMAKE_FLAGS
+                "-DCMAKE_SHARED_LINKER_FLAGS='${CMAKE_SHARED_LINKER_FLAGS}'"
+                ${_rpath_arg}
+              OUTPUT_VARIABLE compile_output)
+
+  set(project_works 1)
+  set(run_output)
+
+  if(project_compiles)
+    execute_process(COMMAND "${test_project_bin_dir}/main"
+                    WORKING_DIRECTORY "${test_project_bin_dir}"
+                    RESULT_VARIABLE project_works
+                    OUTPUT_VARIABLE run_output
+                    ERROR_VARIABLE run_output)
+  endif()
+
+  if(project_works EQUAL 0)
+    set(project_works TRUE)
+    message(STATUS "Performing Test ${testName} - Success")
+  else()
+    set(project_works FALSE)
+    message(STATUS "Performing Test ${testName} - Failed")
+    file(APPEND ${CMAKE_BINARY_DIR}/${CMAKE_FILES_DIRECTORY}/CMakeError.log
+         "Performing Test ${testName} failed with the following output:\n"
+         "BUILD\n-----\n${compile_output}\nRUN\n---\n${run_output}\n")
+  endif()
+
+  set(${projectName} ${project_works} PARENT_SCOPE)
+endfunction()
+
+
+function(has_dynamic_lookup resultVar)
+  # hash the CMAKE_FLAGS passed and check cache to know if we need to rerun
+  string(MD5 cmake_flags_hash "${CMAKE_SHARED_LINKER_FLAGS}")
+
+  if(     NOT DEFINED HAS_DYNAMIC_LOOKUP_hash
+       OR NOT "${HAS_DYNAMIC_LOOKUP_hash}" STREQUAL "${cmake_flags_hash}")
+    unset(HAS_DYNAMIC_LOOKUP)
+  endif()
+
+  if(NOT DEFINED HAS_DYNAMIC_LOOKUP)
+    _test_weak_link_project(has_dynamic_lookup
+                            HAS_DYNAMIC_LOOKUP
+                            DYNAMIC_LOOKUP)
+
+    set(HAS_DYNAMIC_LOOKUP ${has_dynamic_lookup}
+        CACHE BOOL "linker requires \"dynamic_lookup\" for undefined symbols")
+
+    set(HAS_DYNAMIC_LOOKUP_hash "${cmake_flags_hash}"
+        CACHE INTERNAL "hashed flags for HAS_DYNAMIC_LOOKUP check")
+  endif()
+
+  set(${resultVar} "${HAS_DYNAMIC_LOOKUP}" PARENT_SCOPE)
+endfunction()
+
+
+function(has_symbol_dedupe resultVar)
+  # hash the CMAKE_FLAGS passed and check cache to know if we need to rerun
+  string(MD5 cmake_flags_hash "${CMAKE_SHARED_LINKER_FLAGS}")
+
+  if(     NOT DEFINED HAS_SYMBOL_DEDUPE_hash
+       OR NOT "${HAS_SYMBOL_DEDUPE_hash}" STREQUAL "${cmake_flags_hash}")
+    unset(HAS_SYMBOL_DEDUPE)
+  endif()
+
+  if(NOT DEFINED HAS_SYMBOL_DEDUPE)
+    _test_weak_link_project(has_symbol_dedupe
+                            HAS_SYMBOL_DEDUPE
+                            SYMBOL_DEDUPE)
+
+    set(HAS_SYMBOL_DEDUPE ${has_symbol_dedupe}
```

Comment:
Should the cache var be named `LINKER_SUPPORT_SYMBOL_DEDUPE` ? Or  `LINKER_HAS_SYMBOL_DEDUPE`